src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -135,22 +135,37 @@ public ResponseEntity<String> removeTagFromSourceTable(Long uid, String tag) {
 
 	public List<DBTable> searchTablesByTag(String tag) {
 		List<SourceTable> sourcetables = sourceTableRepository.findAll();
-		return sourcetables.stream()
-				.filter(sourcetable -> sourcetable.getTags() != null && sourcetable.getTags().contains(tag))
-				.map(sourcetable -> getDBTable(sourcetable)).collect(Collectors.toList());
+		if (sourcetables.isEmpty()) {
+			throw new CustomException("No tables found in the database.");
+		}
+
+		List<DBTable> result = sourcetables.stream()
+				.filter(sourceTable -> sourceTable.getTags() != null && sourceTable.getTags().contains(tag))
+				.map(sourceTable -> getDBTable(sourceTable)).collect(Collectors.toList());
+		if (result.isEmpty()) {
```

Comment:
Similar to above, It should not be 404. It should be no content. 
Also, there is already a check above if the list is empty, So there is no need to check here.